### PR TITLE
WB-1570: Implement useStyles hook

### PR DESCRIPTION
**Pull request by jandrade**
## Summary:
This hook allows you to use the themed styles in a functional component.

The difference with `useScopedTheme` is that this hook will return the
styles for the component, while `useScopedTheme` will return the theme
object.

### Implementation plan:

1. #1975
2. #1977
3. #1978
4. #1979 
5. **WB-1570: Implement useStyles hook [CURRENT]**
6. #1982
7. Create Storybook example for adding theming support to a custom component (WIP)

Issue: WB-1570

## Test plan:

Navigate to ?path=/docs/theming-usestyles--docs and see the example.

https://5e1bf4b385e3fb0020b7073c-icrhdvimxi.chromatic.com/?path=/docs/theming-usestyles--docs

Screenshot 2023-08-10 at 6 07 11 PM

### CODE REVIEW

File: __docs__/wonder-blocks-theming/use-styles.mdx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+} from "@khanacademy/wonder-blocks-theming";
+
+import {UseStylesExample} from "./theme-examples";
+
+<Meta title="Theming / useStyles" />
+
+# `useStyles`
+
+`useStyles` is a hook that that applies styles based on a given theme.
+
```
Comment (nishasy) [inline __docs__/wonder-blocks-theming/use-styles.mdx:16]: In the PR description, you wrote
> The difference with useScopedTheme is that this hook will return the styles for the component, while useScopedTheme will return the theme object.

Can you add that as a note here in the docs too?

File: packages/wonder-blocks-theming/src/hooks/use-styles.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ *
+ * @param styles The styles to use.
+ * @param theme The theme to be passed to the styles.
+ * @returns The styleSheet object.
+ */
+export default function useStyles<T>(
+    styles: ThemedStylesFn<T>,
+    theme: T,
+): StyleDeclaration {
+    const styleSheet = React.useMemo(() => {
```
Comment (nishasy) [inline packages/wonder-blocks-theming/src/hooks/use-styles.ts:17]: NICE!!!!

File: packages/wonder-blocks-theming/src/hooks/__tests__/use-styles.test.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+                blue: "#0000f0",
+                green: "#00ff00",
+            },
+            spacing: {
+                medium: 8,
+                large: 16,
+            },
+        };
+
+        jest.spyOn(UseScopedTheme, "default").mockReturnValue(theme);
```
Comment (nishasy) [inline packages/wonder-blocks-theming/src/hooks/__tests__/use-styles.test.ts:22]: Is `UseScopedTheme` being used?

File: packages/wonder-blocks-theming/src/hooks/use-styles.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import * as React from "react";
+import {StyleDeclaration, StyleSheet} from "aphrodite";
+
+import {ThemedStylesFn} from "../types";
+
+/**
+ * A hook that that applies styles based on a given theme.
+ *
+ * @param styles The styles to use.
```
Comment (nishasy) [inline packages/wonder-blocks-theming/src/hooks/use-styles.ts:9]: Would it be more accurate to say that this is a function that returns the styles to use?

**Review by nishasy — COMMENTED** [review]
This is really coming along! 👏🏽

**Review by nishasy — APPROVED** [review]

**Comment by jandrade** [comment]
The parent pull-request ([#1979](https://github.com/Khan/wonder-blocks/pull/1979)) has been merged into `main`, but this branch (`theming-styles`) now has conflicts with the new base branch. These conflicts must be resolved before checks can complete on this pull-request.

File: packages/wonder-blocks-theming/src/hooks/__tests__/use-styles.test.ts (hunk 3 above)
Comment (jandrade) [reply packages/wonder-blocks-theming/src/hooks/__tests__/use-styles.test.ts:22]: Nice catch! I'll get rid of it.

**Pull request merged**